libs/langgraph/langgraph/prebuilt/tool_node.py

```diff
@@ -254,10 +282,33 @@ def _run_one(self, call: ToolCall, config: RunnableConfig) -> ToolMessage:
             return invalid_tool_message
 
         try:
-            input = {**call, **{"type": "tool_call"}}
-            tool_message: ToolMessage = self.tools_by_name[call["name"]].invoke(
-                input, config
-            )
+            tool = self.tools_by_name[call["name"]]
+            # check if the Tool.func / Tool._run method returns a Command in the type annotation
+            tool_func = getattr(tool, "func", tool._run)
+            if (return_type := tool_func.__annotations__.get("return")) and (
```

Comment:
why is this code block needed? if we need to update something in langchain core lets do so. we should not be relying on type hints for actual runtime behavior

Reply:
good point -- we're checking the typehint to decide whether we need to get raw tool output (e.g. Command) or a tool message. let me see if it's possible to simplify this